After all, it seems that it is not building on `x86_64` either:

<details>

```
[ 94%] Built target appslib
[ 94%] Built target gcore
[ 94%] Linking CXX shared library libgdal.so

-- stderr output is:
...skipping to end...
            ^
/build/source/src/other/ext/gdal/alg/internal_libqhull/io_r.c: In function 'gdal_qh_printcentrum':
/build/source/src/other/ext/gdal/alg/internal_libqhull/io_r.c:1662:9: note: 'xaxis' declared here
 1662 |   realT xaxis[4], yaxis[4], normal[4], dist;
      |         ^~~~~
In file included from /build/source/src/other/ext/gdal/port/cpl_minixml.h:33,
                 from /build/source/src/other/ext/gdal/gcore/gdal_pam.h:35,
                 from /build/source/src/other/ext/gdal/frmts/ilwis/ilwisdataset.h:32,
                 from /build/source/src/other/ext/gdal/frmts/ilwis/ilwisdataset.cpp:30:
/build/source/src/other/ext/gdal/frmts/ilwis/ilwisdataset.cpp: In member function 'void GDAL::IniFile::Load()':
/build/source/src/other/ext/gdal/port/cpl_port.h:1135:25: warning: attributes at the beginning of statement are ignored [-Wattributes]
 1135 | #define CPL_FALLTHROUGH [[clang::fallthrough]];
      |                         ^~~~~~~~~~~~~~~~~~~~~~
/build/source/src/other/ext/gdal/frmts/ilwis/ilwisdataset.cpp:207:17: note: in expansion of macro 'CPL_FALLTHROUGH'
  207 |                 CPL_FALLTHROUGH
      |                 ^~~~~~~~~~~~~~~
/build/source/src/other/ext/gdal/frmts/ilwis/ilwisdataset.cpp:206:28: warning: this statement may fall through [-Wimplicit-fallthrough=]
  206 |                 s = GetLine(filIni);  // fall through (no break)
      |                     ~~~~~~~^~~~~~~~
/build/source/src/other/ext/gdal/frmts/ilwis/ilwisdataset.cpp:208:13: note: here
  208 |             case FindKey:
      |             ^~~~
In file included from /nix/store/zc0nsv23pakbafngjy32kvhfzb16as43-gcc-13.3.0/include/c++/13.3.0/bits/hashtable.h:35,
                 from /nix/store/zc0nsv23pakbafngjy32kvhfzb16as43-gcc-13.3.0/include/c++/13.3.0/bits/unordered_set.h:33,
                 from /nix/store/zc0nsv23pakbafngjy32kvhfzb16as43-gcc-13.3.0/include/c++/13.3.0/unordered_set:41,
                 from /build/source/src/other/ext/gdal/frmts/vrt/vrtmultidim.cpp:35:
In member function 'std::__detail::_Hash_code_base<_Key, _Value, _ExtractKey, _Hash, _RangeHash, _Unused, __cache_hash_code>::__hash_code std::__detail::_Hash_code_base<_Key, _Value, _ExtractKey, _Hash, _RangeHash, _Unused, __cache_hash_code>::_M_hash_code(const _Key&) const [with _Key = const void*; _Value = const void*; _ExtractKey = std::__detail::_Identity; _Hash = std>
    inlined from 'std::size_t std::__detail::_Hash_code_base<_Key, _Value, _ExtractKey, _Hash, _RangeHash, _Unused, __cache_hash_code>::_M_bucket_index(const std::__detail::_Hash_node_value<_Value, false>&, std::size_t) const [with _Key = const void*; _Value = const void*; _ExtractKey = std::__detail::_Identity; _Hash = std::hash<const void*>; _RangeHash = std::__detail::_>
    inlined from 'std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::size_type std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::_M_bucket_index(const __node_value_type&) const [with _Key = const void*; _Value = const void*; _Alloc = std::allocator<con>
    inlined from 'std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::iterator std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::erase(const_iterator) [with _Key = const void*; _Value = const void*; _Alloc = std::allocator<const void*>; _ExtractKey = st>
    inlined from 'std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::iterator std::_Hashtable<_Key, _Value, _Alloc, _ExtractKey, _Equal, _Hash, _RangeHash, _Unused, _RehashPolicy, _Traits>::erase(iterator) [with _Key = const void*; _Value = const void*; _Alloc = std::allocator<const void*>; _ExtractKey = std::__d>
    inlined from 'std::unordered_set<_Value, _Hash, _Pred, _Alloc>::iterator std::unordered_set<_Value, _Hash, _Pred, _Alloc>::erase(iterator) [with _Value = const void*; _Hash = std::hash<const void*>; _Pred = std::equal_to<const void*>; _Alloc = std::allocator<const void*>]' at /nix/store/zc0nsv23pakbafngjy32kvhfzb16as43-gcc-13.3.0/include/c++/13.3.0/bits/unordered_set.h>
    inlined from 'VRTMDArraySourceFromArray::~VRTMDArraySourceFromArray()' at /build/source/src/other/ext/gdal/frmts/vrt/vrtmultidim.cpp:1852:27:
/nix/store/zc0nsv23pakbafngjy32kvhfzb16as43-gcc-13.3.0/include/c++/13.3.0/bits/hashtable_policy.h:1310:25: warning: potential null pointer dereference [-Wnull-dereference]
 1310 |         return _M_hash()(__k);
      |                ~~~~~~~~~^~~~~
/build/source/src/other/ext/gdal/frmts/grib/degrib/degrib/myerror.cpp: In function 'void AllocSprintf(char**, size_t*, const char*, __va_list_tag*)':
/build/source/src/other/ext/gdal/frmts/grib/degrib/degrib/myerror.cpp:36:27: warning: attributes at the beginning of statement are ignored [-Wattributes]
   36 | #  define CPL_FALLTHROUGH [[clang::fallthrough]];
      |                           ^~~~~~~~~~~~~~~~~~~~~~
/build/source/src/other/ext/gdal/frmts/grib/degrib/degrib/myerror.cpp:245:16: note: in expansion of macro 'CPL_FALLTHROUGH'
  245 |                CPL_FALLTHROUGH
      |                ^~~~~~~~~~~~~~~
/build/source/src/other/ext/gdal/frmts/grib/degrib/degrib/myerror.cpp:36:27: warning: attributes at the beginning of statement are ignored [-Wattributes]
   36 | #  define CPL_FALLTHROUGH [[clang::fallthrough]];
      |                           ^~~~~~~~~~~~~~~~~~~~~~
/build/source/src/other/ext/gdal/frmts/grib/degrib/degrib/myerror.cpp:265:16: note: in expansion of macro 'CPL_FALLTHROUGH'
  265 |                CPL_FALLTHROUGH
      |                ^~~~~~~~~~~~~~~
/build/source/src/other/ext/gdal/frmts/grib/degrib/degrib/myerror.cpp:235:16: warning: this statement may fall through [-Wimplicit-fallthrough=]
  235 |                if ((p1 - p) == 1) {
      |                ^~
/build/source/src/other/ext/gdal/frmts/grib/degrib/degrib/myerror.cpp:246:13: note: here
  246 |             case 'S':
      |             ^~~~
/build/source/src/other/ext/gdal/frmts/grib/degrib/degrib/myerror.cpp:247:16: warning: this statement may fall through [-Wimplicit-fallthrough=]
  247 |                if ((p1 - p) == 1) {
      |                ^~
/build/source/src/other/ext/gdal/frmts/grib/degrib/degrib/myerror.cpp:266:13: note: here
  266 |             default:
      |             ^~~~~~~
/build/source/src/other/ext/gdal/frmts/grib/degrib/degrib/myerror.cpp:269:24: warning: 'char* __builtin___strncpy_chk(char*, const char*, long unsigned int, long unsigned int)' specified bound <unknown> equals destination size [-Wstringop-truncation]
  269 |                strncpy (buffer + ipos, p + 1, p1 - p);
      |                        ^
In file included from /nix/store/4vgk1rlzdqjnpjicb5qcxjcd4spi7wyw-glibc-2.39-52-dev/include/string.h:548,
                 from /build/source/src/other/ext/gdal/port/cpl_port.h:143,
                 from /build/source/src/other/ext/gdal/port/cpl_conv.h:35,
                 from /build/source/src/other/ext/gdal/ogr/ogrsf_frmts/avc/avc.h:118,
                 from /build/source/src/other/ext/gdal/ogr/ogrsf_frmts/avc/avc_e00gen.cpp:93:
In function 'char* strncpy(char*, const char*, size_t)',
    inlined from 'const char* AVCE00GenTableRec(AVCE00GenInfo*, int, AVCFieldInfo*, AVCField*, GBool)' at /build/source/src/other/ext/gdal/ogr/ogrsf_frmts/avc/avc_e00gen.cpp:1411:16:
/nix/store/4vgk1rlzdqjnpjicb5qcxjcd4spi7wyw-glibc-2.39-52-dev/include/bits/string_fortified.h:95:34: warning: 'char* __builtin_strncpy(char*, const char*, long unsigned int)' accessing 1 or more bytes at offsets 0 and [-2147483648, 2147483647] may overlap up to 18446744071562067968 bytes at offset [0, 2147483647] [-Wrestrict]
   95 |   return __builtin___strncpy_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
   96 |                                   __glibc_objsize (__dest));
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/src/other/ext/gdal/ogr/ogrsf_frmts/sqlite/ogrsqlitevirtualogr.cpp:1828:1: warning: missing initializer for member 'sqlite3_module::xIntegrity' [-Wmissing-field-initializers]
 1828 | };
      | ^
/nix/store/wwfrj9kvfi14xclc38qfwm71ah6aawdh-binutils-2.41/bin/ld: final link failed: No space left on device
collect2: error: ld returned 1 exit status
make[5]: *** [CMakeFiles/GDAL.dir/build.make:2232: libgdal.so.32.3.6.3] Error 1
make[4]: *** [CMakeFiles/Makefile2:3597: CMakeFiles/GDAL.dir/all] Error 2
make[3]: *** [Makefile:146: all] Error 2

CMake Error at /build/source/build/src/other/ext/GDAL_BLD-prefix/src/GDAL_BLD-stamp/GDAL_BLD-build-Release.cmake:47 (message):
  Stopping after outputting logs.


make[2]: *** [src/other/ext/CMakeFiles/GDAL_BLD.dir/build.make:86: src/other/ext/GDAL_BLD-prefix/src/GDAL_BLD-stamp/GDAL_BLD-build] Error 1
make[1]: *** [CMakeFiles/Makefile2:13451: src/other/ext/CMakeFiles/GDAL_BLD.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

</details>

--- a/pkgs/applications/graphics/brlcad/default.nix
+++ b/pkgs/applications/graphics/brlcad/default.nix
@@ -1,6 +1,7 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, fetchpatch
 , cmake
 , fontconfig
 , libX11
@@ -11,14 +12,24 @@
 
 stdenv.mkDerivation rec {
   pname = "brlcad";
-  version = "7.34.2";
+  version = "7.38.2";
 
   src = fetchFromGitHub {
     owner = "BRL-CAD";
-    repo = pname;
+    repo = "brlcad";
     rev = "refs/tags/rel-${lib.replaceStrings [ "." ] [ "-" ] version}";
-    hash = "sha256-oafu255xElEIk8p4yvNyR2maykUfxQui/L5MkicA+JA=";
+    hash = "sha256-23UTeH4gY2x/QGYZ64glAkf6LmsXBAppIOHgoUdxgpo=";
   };
+
+  patches = [
+    # This commit was bringing an impurity in the rpath resulting in:
+    # RPATH of binary /nix/store/rq2hjvfgq2nvh5zxch51ij34rqqdpark-brlcad-7.38.0/bin/tclsh contains a forbidden reference to /build/
+    (fetchpatch {
+      url = "https://github.com/BRL-CAD/brlcad/commit/fbdbf042b2db4c7d46839a17bbf4985cdb81f0ae.patch";
+      revert = true;
+      hash = "sha256-Wfihd7TLkE8aOpLdDtYmhhd7nZijiVGh1nbUjWr/BjQ=";
+    })
+  ];
 
   nativeBuildInputs = [
     cmake
@@ -36,11 +47,19 @@ stdenv.mkDerivation rec {
     "-DBRLCAD_ENABLE_STRICT=OFF"
   ];
 
+  env.NIX_CFLAGS_COMPILE = toString [
+    # Needed with GCC 12
+    "-Wno-error=array-bounds"
+  ];
+
   meta = with lib; {
     homepage = "https://brlcad.org";
     description = "BRL-CAD is a powerful cross-platform open source combinatorial solid modeling system";
+    changelog = "https://github.com/BRL-CAD/brlcad/releases/tag/${src.rev}";
     license = with licenses; [ lgpl21 bsd2 ];
     maintainers = with maintainers; [ GaetanLepage ];
     platforms = platforms.linux;
+    # error Exactly one of ON_LITTLE_ENDIAN or ON_BIG_ENDIAN should be defined.
+    broken = stdenv.system == "aarch64-linux";
   };
 }

